### ISSUE TITLE
`df.head(n)` and `df.tail(n)` should show all `n` records

### ISSUE THREAD

**Issue by mcrumiller** [pola-rs/polars#18557]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

If a user supplies the `n` variable to `head`/`tail`, all records should be displayed, not the truncated output.

```pycon
>>> df = pl.DataFrame({"a": pl.int_range(100, eager=True)})
>>> df.head(15)
shape: (15, 1)
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 0   │
│ 1   │
│ 2   │
│ 3   │
│ 4   │
│ …   │ <-- I want to see the rest
│ 10  │
│ 11  │
│ 12  │
│ 13  │
│ 14  │
└─────┘
```

Pandas shows all rows. `head` and `tail` are meant for exploratory analysis and it can be frustrating having to edit `pl.Config` settings to see your data.

### Issue description

a

### Expected behavior

a

### Installed versions

main

**Comment by Julian-J-S** [comment]
I see 2 problems
- sometimes you might actually performa a calculation and then require only the top/bottom X rows (so not only for EDA)
- `head` / `tail` are just normal functions that return a `DataFrame`. Giving them "side-effects" is no nice solution imo

My idea would actually be some kind of `display` method used exclusively for showing your data with lots of formatting options like
- `df.display(rows=..., show_types=..., style=..., ...)`

**Comment by alexander-beedie** [comment]
> My idea would actually be some kind of `display` method used exclusively for showing your data with lots of formatting options like
> 
> * `df.display(rows=..., show_types=..., style=..., ...)`

Yup; I've been thinking of a `show` or `display` method with params that match what you can do with the Config object... We should not special case arbitrary outputs.

**Comment by lmocsi** [comment]
> I like pandas' behavior here, if the dataframe has more than 60 rows, it behaves like polars. But otherwise, it displays the full dataframe. This is why in pandas, if you do `head(50)` you get the full dataframe, it is not because `head` il a "special" function.

The same is with polars, but the default limit is 10. If you say `head(10)` you'll get the full dataframe. A `df.display()` function really would be great here.

**Comment by gab23r** [comment]
I don't think this is the same with polars. 
By default pandas shows around 10 rows for long table and until 60 for small table. 
In polars it is always ten (by default) no matter the size of the dataframe.

**Comment by stinodego** [comment]
This isn't desirable. We need a dedicated `show` function for this.

Closing in favor of https://github.com/pola-rs/polars/issues/16534
